### ISSUE TITLE
linux-next: vdso: Compat vdso helper failed

### ISSUE THREAD

**Issue by avagin** [xemul/criu#273]
https://s3.amazonaws.com/archive.travis-ci.org/jobs/193532880/log.txt
```
========================== Run zdtm/static/env00 in h ==========================
Start test
./env00 --pidfile=env00.pid --outfile=env00.out --envname=ENV_00_TEST
Run criu dump
Run criu restore
Fail
=[log]=> dump/zdtm/static/env00/32/1/restore.log
------------------------ grep Error ------------------------
(00.063057) Error (criu/vdso.c:431): vdso: Compat vdso helper failed
(00.063102) Error (criu/vdso.c:457): vdso: Failed to mmap compatible vdso with helper process
(00.063148) Error (criu/vdso.c:494): vdso: Failed to fill compat vdso symtable
------------------------ ERROR OVER ------------------------
################# Test zdtm/static/env00 FAIL at CRIU restore ##################
```
@0x7f454c46

**Comment by 0x7f454c46** [comment]
Ah-ha, so it's helper process that copies to criu compatible vdso blob to parse (it's kind of optimization, that blob is parsed only once on restore, but not for each restored 32-bit process).
The process, that mmaps 32-bit vdso to copy back to criu died for some reason. 
Hmm-hmm, I tried to reproduce it on  4.10.0-rc4-next-20170120, but no luck.
Could you re-run with attached diff? (github doesn't allow .patch files, so I used .txt)
[diff.txt](https://github.com/xemul/criu/files/719069/diff.txt)

**Comment by 0x7f454c46** [comment]
It shouldn't be anything about env00 test - should be possible to reproduce on any test, so run the full test suite.

**Comment by 0x7f454c46** [comment]
...And it looks like I can't run it on my Travis - some interaction needed?
`linux-next-daily` failed for me, on kexec?
https://travis-ci.org/0x7f454c46/criu/builds/193718723

**Comment by avagin** [comment]
https://travis-ci.org/avagin/criu/builds/193845964

**Comment by 0x7f454c46** [comment]
...so

**Comment by xemul** [comment]
Guys, is this dup for issue #290? Or is it fixed after all?

**Comment by 0x7f454c46** [comment]
Yes, this is a dup. This was the original issue, I've added more debug and some fixes, #290 is  a reproducer.
Let's close this one.
I'll prepare patches and tag them with #290.

**Comment by xemul** [comment]
OK, marking as #290 dup
